### PR TITLE
Microsoft.DigitalTwins version stable/2020-10-31 - Update host value in swagger

### PR DESCRIPTION
**Pull request by johngallardo**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/johngallardo/azure-rest-api-specs/dev-jgalla-update-adt-hostname...master/).

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by briancr-ms — APPROVED** [review]

**Comment by johngallardo** [comment]
The current host property that we set in our swagger specification has been reported as confusing by our documentation team. This change creates a simpler placeholder value, which the customer can replace with the host property which can be found in the portal page for the ADT resource:

image

I looked briefly at using `x-ms-parameterized-host`, but given that this is a GA swagger I was concerned that may introduce a breaking change to the generated clients. Please let me know if that is something I should more strongly consider.

File: specification/digitaltwins/data-plane/Microsoft.DigitalTwins/stable/2020-10-31/digitaltwins.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "title": "Azure Digital Twins API",
     "description": "A service for managing and querying digital twins and digital twin models."
   },
-  "host": "digitaltwins-name.digitaltwins.azure.net",
+  "host": "digitaltwins-hostname",
```
Comment (drwill-ms) [inline specification/digitaltwins/data-plane/Microsoft.DigitalTwins/stable/2020-10-31/digitaltwins.json:8]: @vinagesh do we need to do anything for our test recording/playback to account for this change?

**Review by drwill-ms — APPROVED** [review]

**Comment by johngallardo** [comment]
@lmazuel do you have any idea why the azure-sdk-for-go step is failing? This change seems pretty trivial so I wasn't expecting this failure. @JeffreyRichter maybe you know someone else to follow up with?

**Comment by JeffreyRichter** [comment]
I don't, maybe @jhendrixMSFT does?

**Comment by johngallardo** [comment]
> I don't, maybe @jhendrixMSFT does?

Per @jhendrixMSFT  this appears to be a tooling issue (mishandling empty array vs. null), and should not block merging of this PR.

**Comment by annatisch** [comment]
Hi @johngallardo - just to clarify - when you say the documentation was confusing, is this referring the generated REST API documentation?
I did a quick check of the generated Python SDK using this spec update, and it seems that this value is always overridden by the SDK, nor would be exposed in documentation. While I'm not certain (@zolvarga and @drwill-ms could confirm), I'm pretty sure that should be the case across languages. Is this the expectation?

**Comment by johngallardo** [comment]
> Hi @johngallardo - just to clarify - when you say the documentation was confusing, is this referring the generated REST API documentation?
> I did a quick check of the generated Python SDK using this spec update, and it seems that this value is always overridden by the SDK, nor would be exposed in documentation. While I'm not certain (@zolvarga and @drwill-ms could confirm), I'm pretty sure that should be the case across languages. Is this the expectation?

The issue is in our REST API Reference. E.g.:
https://docs.microsoft.com/en-us/rest/api/digital-twins/dataplane/eventroutes/add

@baanders may be able to provide a bit more context - but I believe that the REST API Reference build simply takes the value from the host header. The goal is to make the value that appears in the documentation to be more accurately reflect the requirements on the customer.

**Comment by baanders** [comment]
Thanks @johngallardo for looking into this.

Yes, the concern is with the value that is displayed in the [REST API reference documentation pages](https://docs.microsoft.com/en-us/rest/api/digital-twins/dataplane/eventroutes/add) John linked above, which readers may use as a resource to craft their own direct API calls. I don't know where it's populated from, aside from that generally these reference docs are generated based on the Swagger-- I just know that from the user perspective, the current documentation suggests use of a value that isn't structured entirely accurately. Assuming that we get this PR change to propagate through to the public reference docs, the rename in this PR resolves the inconsistency.

**Comment by annatisch** [comment]
Thanks @johngallardo, @baanders 
The concern is that the spec is not only used to generate reference docs. It is also used to generate the client SDKs - so we have to be careful not to introduce breaking changes.
As I mentioned, this appears to be fine for the Python SDK. Though would be great to get confirmation from @drwill-ms that it wont impact the other languages :)

**Comment by drwill-ms** [comment]
> As I mentioned, this appears to be fine for the Python SDK. Though would be great to get confirmation from @drwill-ms that it wont impact the other languages :)

My only concern is that it may break our tests after we regenerate, but if it does we'll just fix them, so my concern is not one that would block merging. In fact, I support improving the generated web documentation to prevent customer confusion.

**Comment by johngallardo** [comment]
> > As I mentioned, this appears to be fine for the Python SDK. Though would be great to get confirmation from @drwill-ms that it wont impact the other languages :)
> 
> My only concern is that it may break our tests after we regenerate, but if it does we'll just fix them, so my concern is not one that would block merging. In fact, I support improving the generated web documentation to prevent customer confusion.

@annatisch @JeffreyRichter is there additional action required to merge this?

**Review by annatisch — APPROVED** [review]

**Comment by annatisch** [comment]
Thanks @johngallardo  - nope I think we're good :)

**Pull request merged**